localstack/services/sqs/provider.py

```diff
@@ -791,18 +803,31 @@ def create_queue(
 
         backend = SqsBackend.get(context.region)
 
-        if queue_name in backend.queues:
-            # FIXME #5938: should raise `QueueNameExists` if queue exists with different attributes
-            queue = backend.queues[queue_name]
-            return CreateQueueResult(QueueUrl=queue.url(context))
-        if fifo:
-            queue = FifoQueue(queue_name, context.region, context.account_id, attributes, tags)
-        else:
-            queue = StandardQueue(queue_name, context.region, context.account_id, attributes, tags)
+        with self._mutex:
+            if queue_name in backend.queues:
+                # FIXME #5938: should raise `QueueNameExists` if queue exists with different attributes
+                queue = backend.queues[queue_name]
+                return CreateQueueResult(QueueUrl=queue.url(context))
+
+            if config.SQS_DELAY_RECENTLY_DELETED:
+                deleted = backend.deleted.get(queue_name)
+                if deleted and deleted > (time.time() - RECENTLY_DELETED_TIMEOUT):
+                    raise QueueDeletedRecently(
+                        "You must wait 60 seconds after deleting a queue before you can create "
+                        "another with the same name."
+                    )
+                backend.expire_deleted()
```

Comment:
I'd outdent this one to be called outside the `if config.SQS_DELAY_RECENTLY_DELETED:`, so it would cleanup even when the flag isn't set.